### DISCUSSION TITLE
Présentation du Modèle de Facebook: Prophet

### DISCUSSION THREAD

**Discussion by TWR-RWT** [DstMlOpsCrypto/MainCrypto#21] [Ideas]
### Introduction

Le modèle Prophet, développé par Facebook, est un outil de prévision de séries temporelles conçu pour être simple à utiliser et capable de gérer les séries temporelles avec des composantes saisonnières, des tendances et des jours fériés. Il est particulièrement efficace pour les données présentant des tendances non linéaires avec des saisons annuelles, mensuelles ou hebdomadaires.

### Caractéristiques du Modèle Prophet

1. **Simplicité et Facilité d'Utilisation**: Prophet est conçu pour être utilisé par des personnes ayant peu ou pas d'expérience en prévision de séries temporelles. Il offre une interface simple en Python et R.
  
2. **Gestion des Changements de Tendances**: Prophet permet d'identifier et de gérer les changements de tendances de manière transparente. Cela est particulièrement utile dans les séries temporelles où des ruptures peuvent se produire, comme c'est souvent le cas avec les cours de cryptomonnaies.

3. **Saisonnalité**: Il prend en compte les effets saisonniers, qu'ils soient hebdomadaires, mensuels ou annuels. Cela permet de capturer des schémas récurrents dans les données.

4. **Vacances et Jours Fériés**: Prophet permet d'inclure des jours fériés spécifiques à différents pays, ce qui peut avoir un impact significatif sur certaines séries temporelles.

### Utilisation du Modèle Prophet pour la Prédiction des Cours de Cryptomonnaies

L'utilisation de Prophet peut offrir certains avantages pour la prévision des cours de cryptos :

1. **Identification des Tendances**: Prophet peut aider à identifier des tendances à long terme, permettant de mieux comprendre la direction générale du marché.

2. **Gestion de la Saisonnalité**: Les cryptomonnaies peuvent montrer des schémas saisonniers (comme des hausses de prix lors de certaines périodes de l'année, ex "sale in may go away"). Prophet permet de capturer ces schémas et de les intégrer dans les prévisions.

3. **Modélisation des Changements Brutaux**: Les marchés des cryptomonnaies peuvent subir des changements brusques dus à des nouvelles ou des événements économiques. Prophet permet de modéliser ces changements et de les intégrer dans les prévisions futures.

### Études de Cas et Exemples

1. **[Prévision des Prix du Bitcoin avec ARIMA, XGBoost, Prophet et LSTM](https://github.com/AaronFlore/Forecasting-Bitcoin-Prices/blob/main/forecasting-btc-arima-xgboost-prophet-lstm.ipynb)**: Ce repository github donne plusieurs exemple de différent modèles nous permettant de les comparer. Dans cet étude Prophet  semble assez décevant (prédictions tres éloignées de la réalité) bien qu'il semble arrivé à identifier des périodes "fortes" et des périodes "faibles". En revanche nous trouvons aussi dans l'étude un modèle LSTM qui a l'air de donner de bonnes prédictions. 

2. **[Analyse et Prédiction des Prix de l'Ethereum](https://www.kaggle.com/code/kaushiksuresh147/ethereum-eda-and-prediction-using-prophet)**: Cette étude utilise Prophet pour l'analyse exploratoire des données (EDA) et la prévision des prix de l'Ethereum. Elle démontre comment Prophet peut être appliqué pour obtenir des prévisions efficaces et explique les étapes de préparation des données, d'entraînement du modèle et d'évaluation des performances. L'étude est vraiment bien expliquée et peut être suivi comme un petit tuto.

3. **[Bitcoin Price Prediction with FB Prophet](https://www.youtube.com/watch?v=da5GNkAZO54)**:
Vidéo présentant un cas d'utilisation de Prophet pour prédire les cours du bitcoin. La vidéo explique bien les différentes étapes et le fonctionnement de Prophet. Les résultats sont cependant décevant.

### Conclusion

Le modèle Prophet de Facebook est un outil puissant et flexible pour la prévision de séries temporelles. Sa capacité à gérer les tendances non linéaires, la saisonnalité et les événements spécifiques le rend intéressant au premier abord. De plus gère les time séries non stationnaire (comme les cours de cryptomonnaies) contrairement à des modèles comme ARIMA. Cependant dans les études que j'ai pu voir utilisant Prophet pour prédire les cours de cryptos, les résultats étaient plutôt décevant contrairement à des modèles LSTM.

**Comment by mis7ern9o** [comment]
@TWR-RWT tu peux fermer (close) la discussion au besoin pour rappel)
